tutorials/define_and_compile_model_webgl.py

```diff
@@ -0,0 +1,289 @@
+To run this notebook, you need to install tvm and nnvm following
+`these instructions <https://github.com/dmlc/nnvm/blob/master/docs/how_to/install.md>`_.
+Notice that you need to build tvm with OpenGL.
+"""
+
+######################################################################
+# Overview for Supported Hardware Backend of TVM
+# ----------------------------------------------
+# The image below shows hardware backend currently supported by TVM:
+#
+# .. image:: https://github.com/dmlc/web-data/raw/master/tvm/tutorial/tvm_support_list.png
+#      :align: center
+#      :scale: 100%
+#
+# In this tutorial, we'll choose OpenGL as the target backend.
+# To begin with, let's import NNVM and TVM.
+from __future__ import print_function
+import tvm
+import nnvm.compiler
+import nnvm.testing
+
+# To run the RPC demo, set this flag to True.
+use_rpc = False
+
+# To run the WebGL deploy demo, set this flag to True.
+use_deploy = True
+
+######################################################################
+# Define Neural Network in NNVM
+# -----------------------------
+# First, let's define a neural network with nnvm python frontend.
+# For simplicity, we'll use pre-defined resnet network in NNVM.
+# Parameters are initialized with Xavier initializer.
+# NNVM also supports other model formats such as MXNet, CoreML and ONNX.
+#
+# In this tutorial, we assume we will do inference on our device
+# and the batch size is set to be 1. Input images are RGB color
+# images of size 224 * 224. We can call the :any:`nnvm.symbol.debug_str`
+# to show the network structure.
+
+batch_size = 1
+num_class = 1000
+image_shape = (3, 224, 224)
+data_shape = (batch_size,) + image_shape
+out_shape = (batch_size, num_class)
+
+net, params = nnvm.testing.resnet.get_workload(
+    batch_size=batch_size, image_shape=image_shape)
+print(net.debug_str())
+
+######################################################################
+# Compilation
+# -----------
+# Next step is to compile the model using the NNVM/TVM pipeline.
+# Users can specify the optimization level of the compilation.
+# Currently this value can be 0 to 2, which corresponds to
+# "SimplifyInference", "OpFusion" and "PrecomputePrune" respectively.
+# In this example we set optimization level to be 0
+# and use OpenGL as compile target.
+#
+# :any:`nnvm.compiler.build` returns three components: the execution graph in
+# json format, the TVM module library of compiled functions specifically
+# for this graph on the target hardware, and the parameter blobs of
+# the model. During the compilation, NNVM does the graph-level
+# optimization while TVM does the tensor-level optimization, resulting
+# in an optimized runtime module for model serving.
+#
+# To generate the module library, TVM will first transfer graph IR into lower
+# intrinsic IR for the specified target backend, which is OpenGL in this
+# example. Then target backend will generate module library.
+
+if not tvm.module.enabled("opengl"):
```

Comment:
a bit too many opencl checks, maybe we can just use exit(0) here

Reply:
add a comment in the block, saying that because this tutorial is built in CI server, so the result may not be displayed, but user can download the notebook and run it

Reply:
Now only check once.